### ISSUE TITLE
[ENHANCEMENT] Allow Spaces in Internal Link Auto-Complete

### ISSUE THREAD

**Issue by varian93** [Zettlr/Zettlr#622]
## Description
Allow spaces in the [[Internal Link]] auto-complete feature.

## Proposed Changes
When using the [[Internal Link]] feature it shows potential auto-complete candidates through a drop down menu. Currently any spaces are "not allowed" & "break" it. 

If I have a note "20200319070147 - an idealist believes.md" that I want to link to from another note & I begin to type "[[an" it will show a potential match of "20200319070147 - an idealist believes" which is great, but as soon as I type the space between "an" and "idealist" it shows no potential matches.

I would find it much more useful if it would allow the space so I could continue to narrow down the possibilities, as I may have many files that start with "an". 

When writing notes I find it much easier to remember part of a potential note to link to's title than I do any part of the timestamp ID. 

## Caveats
None come to mind.

**Comment by nathanlesage** [comment]
I deliberately allowed spaces to close the autocomplete, precisely because one of the most annoying features of IDEs such as VSCode or Atom is that you cannot close them with space, but have to always explicitly hit Escape, sometimes multiple times (depending on the amount of dropdowns currently open — which is a problem of VSCode specifically 😆). This is a total dealbreaker for me, which is why I deliberately enabled the Space to close autocomplete. Normally, a specific word is enough, you don't have to do all of them!

In case you really want this, please feel free to comment (I'll see all your comments), and propose a possibility which would keep space closing autocomplete for most users and a workaround to add spaces!

**Comment by varian93** [comment]
I understand your position. 

The easiest most accessible solution that I've been able to think of is a user setting in the preferences under the "Zettelkasten" tab. It is there that the user is able to choose what characters begin/end (open/close) the internal linking feature. Thus it seems like the first place a user would likely look when wanting to change a behavior related to this aspect of the application.

Perhaps somewhere nearby on that same page there could be an "Internal Link Autocomplete Behavior" section where the user would be able to choose if a space would close the autocomplete or not.

Perhaps it could be combined in the "When linking files, add the filename..." section.

The other idea that I had was a modifier key that could be used when entering a "space" that one explicitly did not want to close the autocomplete (e.g. shift+space). The issue that I took with that approach was that it would be difficult for a user to know/find (and remember) without having to refer to the documentation. Where as with the setting in the preferences they would see that it could be set to their liking. I also realized that most people would either want it one way or the other. 

The 2 ideas could be combined. The setting in the preferences would determine the default behavior which could be overridden on a per use basis with the use of a modifier+space key sequence.
